packages/typespec-vscode/src/extension.ts

```diff
@@ -18,27 +26,60 @@ export async function activate(context: ExtensionContext) {
   context.subscriptions.push(createTaskProvider());
 
   context.subscriptions.push(
-    commands.registerCommand("typespec.showOutputChannel", () => {
+    commands.registerCommand(CommandName.ShowOutputChannel, () => {
       outputChannel.show(true /*preserveFocus*/);
     }),
   );
 
   context.subscriptions.push(
-    commands.registerCommand("typespec.restartServer", async () => {
-      if (client) {
-        await client.restart();
-      }
+    commands.registerCommand(
+      CommandName.RestartServer,
+      async (args: RestartServerCommandArgs | undefined): Promise<TspLanguageClient> => {
+        return vscode.window.withProgress(
+          {
+            title: "Restarting TypeSpec language service...",
+            location: vscode.ProgressLocation.Notification,
+          },
+          async () => {
+            if (args?.forceRecreate === true) {
+              logger.info("Forcing to recreate TypeSpec LSP server...");
+              return await recreateLSPClient(context, args?.popupRecreateLspError);
+            }
+            if (client && client.state === State.Running) {
+              await client.restart();
+              return client;
+            } else {
+              logger.info(
+                "TypeSpec LSP server is not running which is not expected, try to recreate and start...",
+              );
+              return await recreateLSPClient(context, args?.popupRecreateLspError);
```

Comment:
Nit: You don't need to `return await` unless you're trying to do something like catch an error thrown by the thing you're awaiting.

Reply:
Since this is an async function though, whatever you return is wrapped in a Promise (unless it is already a promise). So this is essentially unwrapping the promise returned by this function call, then immediately wrapping it in a new promise.

Not a big deal - it's a minor optimization and the code will run correctly, was just pointing it out.

Reply:
get it. makes sense. code updated. thanks.